.github/workflows/code-quality.yaml

```diff
@@ -0,0 +1,52 @@
+name: Code Quality Checks
+on:
+  push:
+    branches:
+      - dev
+      - main
+      - release/**
+  pull_request:
+    branches:
+      - dev
+      - main
+      - release/**
+  workflow_call:
+  workflow_dispatch: {}
+defaults:
+  run:
+    working-directory: .
+jobs:
+  code-quality:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python_version:
+          - "3.7"
+          - "3.8"
+          - "3.9"
+        pip_deps:
+          - "[all]"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Cache pip
```

Comment:
How important is the cache? I'd like to catch version changes even if `setup.py` didn't change. e.g. we didn't pin a dep that we should have.

Reply:
That's a good point, it's mostly just a runtime improvement.  I'll remove it for now and we can consider adding it back if it's considerably slower.